### PR TITLE
chore: Check docs module dependencies and bundle size

### PR DESCRIPTION
**Pull request by salarenko**
#### Please provide a link to the associated issue.
Closes: #2854 

#### Please provide a brief summary of this pull request.
**Why our build previews bundles weights up to 11MB?**

For preview builds we are using unoptimized version of production build (no optimization flags, no gzip etc.) all of this to reduce the pipeline time. Properly processed production builds are deployed as we officially release new versions of our documentation.

Properly processed bundle:
image

I think this clarifies that we don't have an issue with our production documents/library bundle weight and explains the difference in size between netlify preview and production bundle size.

Complete production build stats:
[stats.zip](https://github.com/SAP/fundamental-ngx/files/5033779/stats.zip)

**This PR:**
I've checked dependencies between the modules and cleaned up shareable modules, so they provide only needed dependencies.
I've splitted shared module into:
- `SharedDocumentationModule` - Provides dependencies needed to build documentation shell
- `SharedDocumentationPageModule` - Provides dependencies needed to build component documentation page
and updated/cleaned up **Core and Platform** documentation modules.

**Files worth looking at:**
- `apps/docs/src/app/documentation/shared-documentation.module.ts`
- `apps/docs/src/app/documentation/shared-documentation-page.module.ts`

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Comment by InnaAtanasova** [comment]
LGTM. Won't approve to give a chance to @fkolar and @KevinOkamoto to take a look at the Platform changes.

**Comment by droshev** [comment]
> There are a few conflicts that need to be resolved, but other than that it looks good. Resolve conflicts and I'll review again.

@KevinOkamoto can you recheck again?

File: apps/docs/src/app/core/core-documentation.module.ts [hunk 1]
```diff
@@ -16,7 +16,6 @@ import {
     SELECT_MOBILE_CONFIG
 } from '../documentation/utilities/consts';
 
-
```
Comment (stefanoScalzo) [inline apps/docs/src/app/core/core-documentation.module.ts:19]: remove file

File: apps/docs/src/app/documentation/core-helpers/sections-toolbar/sections-toolbar.component.ts [hunk 2]
```diff
@@ -5,8 +5,8 @@ import { BehaviorSubject } from 'rxjs';
 const SMALL_SCREEN_BREAKPOINT = 992;
 @Component({
     selector: 'sections-toolbar',
-    styleUrls: ['./sections-toolbar.component.scss'],
-    templateUrl: './sections-toolbar.component.html'
+    templateUrl: './sections-toolbar.component.html',
+    styleUrls: ['./sections-toolbar.component.scss']
```
Comment (stefanoScalzo) [inline apps/docs/src/app/documentation/core-helpers/sections-toolbar/sections-toolbar.component.ts:9]: formatting shouldnt be in this pr

File: apps/docs/src/app/core/core-documentation.module.ts (hunk 1 above)
Comment (salarenko) [reply apps/docs/src/app/core/core-documentation.module.ts:19]: Could you be more specific what is the problem?

**Comment by salarenko** [comment]
@KevinOkamoto @droshev 
PR has been updated. Please find a moment to check it so we can merge.
To check this PR I recommend going thorough documentation pages and see if they load properly (they do 😛) and check that dependencies are brought by `SharedDocumentationModule` and `SharedDocumentationPageModule` (more in PR description).

**Comment by salarenko** [comment]
@stefanoScalzo I think these are very details of this PR and there is no point of moving them into separate PR. Fixing small inconsistencies like formatting is fine until it doesn't cover actual PR purpose.

**Pull request merged**